tools/apps/src/bin/quiche-server.rs

```diff
@@ -472,6 +475,19 @@ fn main() {
                 }
 
                 trace!("{} written {} bytes", client.conn.trace_id(), write);
+
+                // limit write bursting
+                client.bytes_sent += write;
+
+                if client.bytes_sent >= MAX_SEND_BURST_LIMIT {
+                    trace!(
+                        "{} pause writing at {}",
+                        client.conn.trace_id(),
+                        client.bytes_sent
+                    );
+                    client.bytes_sent = 0;
+                    break;
```

Comment:
My understanding is that after the break, the server will block again on the event loop (epoll/...). So in order for packets to be sent out again from a connection that reached the limit, a new I/O event (or a timeout) would need to happen, which seems like it could potentially stall a connection, or even completely hang it (e.g. if no event happens anymore, due to idk, packet loss on the other side of the connection).

So I'm not sure this is the proper way to resolve the problem, but I also can't think of an alternative right now.

Reply:
Right it can happen. I added some trick to set poll timeout to 0 when we need to continue writing, not to wait for incoming packet.